File: 938-RangeSumOfBST/Solution.java
/**
 * 938. Ramge Sum of BST
 * Jun. 22, Google NYC
 * 
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int L, int R) {
        if (root == null || L > R) {
            return 0;
        }
        int val = root.val;
        int res = 0;
        int sum = rangeSumBST(root.left, L, R) + rangeSumBST(root.right, L, R);
        if (val >= L && val <= R) {
            sum += val;
        }
        return sum;
    }
}

File: 938-RangeSumOfBST/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        if not root:
            return 0
        v = root.val
        if low <= v <= high:
            return v + self.rangeSumBST(root.left, low, high) \
                     + self.rangeSumBST(root.right, low, high)
        elif v < low:
            return self.rangeSumBST(root.right, low, high)
        else:
            return self.rangeSumBST(root.left, low, high)
